.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,52 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+{
+  "name": "Torchgeo DevContainer",
+  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+  "image": "mcr.microsoft.com/devcontainers/python:3.12",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    "ghcr.io/devcontainers/features/git:1": { "version": "os-provided" }
+  },
+
+  // "postCreateCommand": "",
+  "postStartCommand": "pip install -e .[all]",
+
+  // Enable GPU support
+  "hostRequirements": {
+    "gpu": "optional"
+  },
+
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "ms-python.python",
+        "ms-python.isort",
+        "ms-python.black-formatter",
+        "ms-python.vscode-pylance",
+        "ms-python.pylint",
```

Comment:
Do we need these? Not sure if I want to enforce any style constraints that aren't already enforced in CI.

Reply:
That's extension, I would keep them espectially pylance. This give many IDE feedback to the user even before the CI. For example the current settings will display pylint error on the IDE side before any CI. It will also auto format the code with the same specification as the CI. This may put less work on the CI side and enforce best practice directly in the IDE

Reply:
Yes, I definitely want to keep black/mypy/isort/pydocstyle/flake8 because we also enforce that in CI. I'm specifically commenting on things like pylance and pylint which we don't use.

Reply:
You can remove pylint. I would keep pylance anyway because it's just visual, it doesn't enforce anything. 
Btw for torchgeo in general you can have a look to [ruff](https://docs.astral.sh/ruff/) that bundle everything in a single package.

Reply:
See https://github.com/microsoft/torchgeo/pull/1908#issuecomment-1964000416 for my thoughts on ruff. Not strongly opposed in principle, but let's discuss that in a separate issue.

Reply:
Does pylance flag anything that the other tools don't flag?